source/material_model/visco_plastic.cc

```diff
@@ -715,29 +738,91 @@ namespace aspect
+                             "and its second invariant is used to weaken both the plastic viscosity "
+                             "(specifically, the cohesion and yield angle) and the viscous viscosity "
+                             "that arises from diffusion and/or dislocation creep."
+                             "\n\n"
+                             "\\item ``total strain'': Only the second invariant of the strain tensor is "
+                             "tracked, and it is used to weaken both the plastic viscosity "
+                             "(specifically, the cohesion and yield angle) and the viscous viscosity."
+                             "\n\n"
+                             "\\item ``plastic weakening with plastic strain only'': The second invariant "
+                             "of the strain accumulated by plastic deformation is tracked, and is used "
+                             "to weaken the plastic viscosity (specifically, the cohesion and yield angle). "
+                             "The viscous viscosity is not weakened."
+                             "\n\n"
+                             "\\item ``plastic weakening with total strain only'': The second invariant of "
+                             "the strain tensor is tracked, and it is used to weaken the plastic viscosity "
+                             "(specifically, the cohesion and yield angle). "
+                             "The viscous viscosity is not weakened."
+                             "\n\n"
+                             "\\item ``plastic weakening with plastic strain and viscous weakening with viscous strain'': "
+                             "Both the strain accumulated by plastic deformation and by viscous deformation are "
+                             "computed separately. The plastic strain is used to weaken the plastic viscosity "
+                             "(specifically, the cohesion and yield angle); the viscous strain is used to "
+                             "weaken the viscous viscosity. "
+                             "\n\n"
+                             "\\item ``viscous weakening with viscous strain only'': The second invariant of "
+                             "the strain accumulated by viscous deformation is tracked, and is used to "
+                             "weaken the viscous viscosity. The plastic viscosity is not weakened."
+                             "\n\n"
+                             "\\item ``default'': The default option has the same behavior as ``none'', "
+                             "but is there to make sure that the original parameters for specifying the "
+                             "strain weakening mechanism (``Use plastic/viscous strain weakening'') are still allowed, "
+                             "but to guarantee that one uses either the old parameter names or the new ones, "
+                             "never both.");
+          prm.declare_entry ("Use strain weakening", "default",
```

Comment:
Given the amount of code required to keep backward compatibility, I'm inclined towards outright removal. However, that can always be done in a follow-up PR as it may be quite a bit of work.

Reply:
@naliboff, what do you think how many people use this at the moment? The feature was introduced before the last release (the changelog says 2017/04/14) and I am not sure we want to break people's prm files again without even marking the option as deprecated...

Reply:
On second thought, I agree. A fair number of people are likely using the feature by now, so probably best to not break functionality until we get close to the next major release.